goodkey/blocked.go

```diff
@@ -10,30 +13,42 @@ import (
 	yaml "gopkg.in/yaml.v2"
 )
 
-// blockedKeys is a type for maintaining a map of Base64 encoded SHA256 hashes
+type keyHash [32]byte
+
+// blockedKeys is a type for maintaining a map of SHA256 hashes
 // of SubjectPublicKeyInfo's that should be considered blocked.
 // blockedKeys are created by using loadBlockedKeysList.
-type blockedKeys map[string]bool
+type blockedKeys map[keyHash]bool
+
+var ErrWrongDecodedSize = errors.New("not enough bytes decoded for sha256 hash")
 
 // blocked checks if the given public key is considered administratively
-// blocked based on a Base64 encoded SHA256 hash of the SubjectPublicKeyInfo.
+// blocked based on a SHA256 hash of the SubjectPublicKeyInfo.
 // Important: blocked should not be called except on a blockedKeys instance
 // returned from loadBlockedKeysList.
 // function should not be used until after `loadBlockedKeysList` has returned.
 func (b blockedKeys) blocked(key crypto.PublicKey) (bool, error) {
-	hash, err := core.KeyDigest(key)
+	b64Hash, err := core.KeyDigest(key)
 	if err != nil {
 		// the bool result should be ignored when err is != nil but to be on the
 		// paranoid side return true anyway so that a key we can't compute the
 		// digest for will always be blocked even if a caller foolishly discards the
 		// err result.
 		return true, err
 	}
+	var hash keyHash
```

Comment:
Rather than doing this encoding/decoding dance in KeyDigest and here, can we instead split KeyDigest into KeyDigest, which returns a [32]byte, and Base64KeyDigest, which returns a b64 string (or something).